translation/translation.js

```diff
@@ -0,0 +1,271 @@
+                            $("div#content div#map-content div#map.olMap div#OpenLayers_Map_377_OpenLayers_ViewPort div#OpenLayers_Map_377_OpenLayers_Container div#liz_layer_popup.olPopup.lizmapPopup div#liz_layer_popup_GroupDiv div#liz_layer_popup_contentDiv.olPopupContent.lizmapPopupContent div.lizmapPopupDiv table.lizmapPopupTable tbody tr th").filter(function(){
+                                return $(this).text() === italian
+                            }).html(translationIT_FR3[italian]);
+                        }
+                        else {
+                            $("div#content div#map-content div#map.olMap div#OpenLayers_Map_377_OpenLayers_ViewPort div#OpenLayers_Map_377_OpenLayers_Container div#liz_layer_popup.olPopup.lizmapPopup div#liz_layer_popup_GroupDiv div#liz_layer_popup_contentDiv.olPopupContent.lizmapPopupContent h4").filter(function(){
+                                return $(this).text() === italian
+                            }).html(italian);
+                            $("div#content div#map-content div#map.olMap div#OpenLayers_Map_377_OpenLayers_ViewPort div#OpenLayers_Map_377_OpenLayers_Container div#liz_layer_popup.olPopup.lizmapPopup div#liz_layer_popup_GroupDiv div#liz_layer_popup_contentDiv.olPopupContent.lizmapPopupContent div.lizmapPopupDiv table.lizmapPopupTable tbody tr th").filter(function(){
+                                return $(this).text() === italian
+                            }).html(italian);
+                        }
+                    });
+            });
+        }
+    }
+});
+
+//This function is executed when the minidockopened event is triggered (it works only if the location tool is opened clicking on the related button of the toolbar)
+lizMap.events.on({
+    minidockopened: function(){
+        var userLang = navigator.language || navigator.userLanguage;
+        if (userLang === 'fr' || userLang === 'fr-FR'){
+            // translate the layer name shown in the location tool 
+            $("div#content div#map-content div#mini-dock div.tabbable.tabs-below div#mini-dock-content.tab-content div#locate.tab-pane.active div.locate div.menu-content div.locate-layer span.custom-combobox input[placeholder='Comuni Roia']").attr("placeholder", "Vive la revolucion");
+        }
+    }
+});
+
+//This function is executed when the lizmapswitcheritemselected event is triggered
+lizMap.events.on({
+    lizmapswitcheritemselected: function(){
+        var userLang = navigator.language || navigator.userLanguage;
+        if (userLang === 'fr' || userLang === 'fr-FR'){        
+            
+            var translationIT_FR = {};
+
+            // get the json file with transalted strings
+            // the path to the json file must be provided as relative path, using the web url the browser can return a security error
+            fetch('../../../../../../concerteaux3d/translation.json').then(function(response) {
```

Comment:
You could cache translation.js as you have download it with "uicreated" event above.

Reply:
I have changed the code, now the fetch of the translation.json file is done only once in the "uicreated" event.